src/components/MyObservations/MyObservationsHeader.js

```diff
@@ -0,0 +1,121 @@
+// @flow
+import { useNavigation } from "@react-navigation/native";
+import MyObservationsToolbar from "components/MyObservations/MyObservationsToolbar";
+import { Button, Heading1, Subheading1 } from "components/SharedComponents";
+import { View } from "components/styledComponents";
+import type { Node } from "react";
+import React, { useEffect, useRef } from "react";
+import { Trans, useTranslation } from "react-i18next";
+import { Animated } from "react-native";
+import User from "realmModels/User";
+import useNumUnuploadedObservations from "sharedHooks/useNumUnuploadedObservations";
+
+import MyObservationsOnboarding from "./MyObservationsOnboarding";
+
+type Props = {
+  setLayout: Function;
+  layout: string,
+  hideHeaderCard: boolean,
+  currentUser: ?Object,
+  hideToolbar: boolean,
+  numOfObservations: number,
+  setHeightAboveToolbar: Function
+}
+
+const fade = ( value, duration ) => ( {
+  toValue: value,
+  duration,
+  useNativeDriver: true
+} );
+
+const MyObservationsHeader = ( {
+  setLayout,
+  layout,
+  hideHeaderCard,
+  currentUser,
+  hideToolbar,
+  numOfObservations,
+  setHeightAboveToolbar
+}: Props ): Node => {
+  const fadeAnimation = useRef( new Animated.Value( 0 ) ).current;
+  const navigation = useNavigation( );
+  const numUnuploadedObs = useNumUnuploadedObservations( );
```

Comment:
I think I'm feeling a little confused about what our goal is. If the goal is testability, then I'm definitely happy to pull this into a parent component. I thought one of our goals was to use hooks so that we didn't have to pass props to multiple children components -- at least that's why I've been writing hooks and using them closest to where they're needed.

Reply:
Well, like I said, I'm on the fence in this specific case, particularly given that the component is not reusable, so modularity is not a concern. Here I'd say you should make the call.

In general, though, trying to make components as stateless as possible helps separate concerns by keeping presentation code separate for business logic. React has no formal equivalent to a controller or a view controller that would usually be the home of non-presentational logic, so I'm struggling to understand how to achieve that separation. On the web we use Redux which adds the "container" concept that serves this purpose: it's an interface between a presentational component and state. Hooks sort of play this role by literally moving some state code away from the presentational code, but I feel like they also work against modularity by making components less predictable with async effects.